lib/pure/ioselects/ioselectors_kqueue.nim

```diff
@@ -375,8 +375,8 @@ proc unregister*[T](s: Selector[T], fd: int|SocketHandle) =
            "Descriptor [" & $fdi & "] is not registered in the queue!")
 
   if pkey.events != {}:
-    if pkey.events * {Event.Read, Event.Write} != {}:
-      if Event.Read in pkey.events:
+    if pkey.events * {Event.Read, Event.Write, Event.User} != {}:
```

Comment:
Please check one more time all the `elif` branches in this procedure, you will find your changes in line 422.

Reply:
Good catch!
Pushing an update to remove this.

Note this validates my changes to the other implementations, because both `ioselectors_select.nim` and `ioselectors_epoll.nim` do not account for this case!
`ioselectors_poll.nim` works because it does not branch based on the `pkey.Event` type.

Reply:
@cheatfate can you give a final word on this? I believe the `kqueue` implementation shows that this is a valid bug fix, but I would like to get your final thoughts.

specifically, my understanding is as follows:

- For the `FD` / `SocketHandle` variant of `unregister`:
  - `ioselectors_kqueue.nim` handles the `Event.User` case 
  - `ioselectors_epoll.nim` and Windows `ioselectors_select.nim` **do not** handle the `Event.User` case in the equivalent `unregister` code.
  - This is incorrect behavior (that is fixed with this PR)